webapp/channels/src/components/admin_console/elasticsearch_settings.tsx

```diff
@@ -36,7 +56,7 @@ export default class ElasticsearchSettings extends AdminSettings {
         return config;
     };
 
-    getStateFromConfig(config) {
+    getStateFromConfig(config: AdminConfig) {
```

Comment:
https://github.com/mattermost/mattermost/blob/8056be1d165c68e19dcda7429001055af9489d75/webapp/channels/src/components/admin_console/bleve_settings.tsx#L23-L25

Can you do similar to this. So it all ties back to Props